new-pipeline/src/dashboards.js

```diff
@@ -149,67 +167,116 @@ function loadStateFromUrlAndCookie() {
+    var aggregates = pageState.aggregates.split("!").filter(function(v) {
+      return [
+        "5th-percentile", "25th-percentile", "median",
+        "75th-percentile", "95th-percentile", "mean"
+      ].indexOf(v) >= 0;
+    });
     if (aggregates.length > 0) { pageState.aggregates = aggregates; }
     else { pageState.aggregates = ["median"]; }
   } else { pageState.aggregates = ["median"]; }
-  pageState.measure = typeof pageState.measure === "string" && pageState.measure !== "" && pageState.measure !== "null" ? pageState.measure : "GC_MS";
-  pageState.min_channel_version = typeof pageState.min_channel_version === "string" && pageState.min_channel_version.indexOf("/") >= 0 ?
-    pageState.min_channel_version : null;
-  pageState.max_channel_version = typeof pageState.max_channel_version === "string" && pageState.max_channel_version.indexOf("/") >= 0 ?
-    pageState.max_channel_version : null;
-  pageState.product = typeof pageState.product === "string" && pageState.product !== "" && pageState.product !== "null" ?
-    pageState.product.split("!").filter(function(v) { return v !== ""; }) : ["Firefox"];
-  pageState.os = typeof pageState.os === "string" && pageState.os !== "" && pageState.os !== "null" ?
-    pageState.os.split("!").filter(function(v) { return v !== ""; }) : null;
-  pageState.arch = typeof pageState.arch === "string" && pageState.arch !== "" && pageState.arch !== "null" ?
-    pageState.arch.split("!").filter(function(v) { return v !== ""; }) : null;
-  pageState.e10s = typeof pageState.e10s === "string" && pageState.e10s !== "" && pageState.e10s !== "null" ?
-    pageState.e10s.split("!").filter(function(v) { return v !== ""; }) : null;
-  pageState.processType = typeof pageState.processType === "string" && pageState.processType !== "" && pageState.processType !== "null" ?
-    pageState.processType.split("!").filter(function(v) { return v !== ""; }) : null;
-  pageState.compare = typeof pageState.compare === "string" && ["", "os", "osVersion", "architecture", "e10sEnabled", "child"].indexOf(pageState.compare) >= 0 ?
-    pageState.compare : "";
-  
+  pageState.measure = (
+    typeof pageState.measure === "string" &&
+    pageState.measure !== "" && pageState.measure !== "null"
+  ) ? pageState.measure : "GC_MS";
+  pageState.min_channel_version = (
+    typeof pageState.min_channel_version === "string" &&
+    pageState.min_channel_version.indexOf("/") >= 0
+  ) ? pageState.min_channel_version : null;
+  pageState.max_channel_version = (
+    typeof pageState.max_channel_version === "string" &&
+    pageState.max_channel_version.indexOf("/") >= 0
+  ) ?  pageState.max_channel_version : null;
+  pageState.product = (
+    typeof pageState.product === "string" &&
+    pageState.product !== "" && pageState.product !== "null"
+  ) ? pageState.product.split("!").filter(function(v) { return v !== ""; }) : ["Firefox"];
```

Comment:
the over-use of the conditional operator is making the code super-dense and hard to read, replace with ifs whenever it's a statement more complex than "x = (a + 2 > b) ? 1 : 0"